### PR TITLE
feat(endpoints-overlay): Add i18n support.

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
This PR adds i18n support to the `endpoints-overlay` component and converts the code to TypeScript.

### CODE REVIEW

File: packages/endpoints-overlay/src/endpoint.tsx [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+  swapLocation = (): void => {
+    const { location, setLocation, type } = this.props;
+    this.clearLocation();
+    const otherType = type === "from" ? "to" : "from";
+    setLocation({ locationType: otherType, location });
+  };
+
+  onDragEnd = e => {
+    const { setLocation, type } = this.props;
+    // This method is depcreated. the latlng object should be fed into react intl
```
Comment (philip-cline) [inline packages/endpoints-overlay/src/endpoint.tsx:128]: *deprecated

**Review by philip-cline — APPROVED** [review]
Just some nit picky formatting points, otherwise looks & works good for me!

File: packages/endpoints-overlay/src/EndpointsOverlay.story.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  name: "Portland Towers",
+  type: "home"
+};
+const locations = [fromLocation, toLocation];
+
+function CatDogIcon({ type }: UserLocationAndType) {
+  return type === "from" ? <Cat size={40} color="orange" /> : <Dog size={40} />;
+}
+
+const mapDecorator = (
```
[the remark this answers is not in the document]
Comment (binh-dam-ibigroup) [reply packages/endpoints-overlay/src/EndpointsOverlay.story.tsx:40]: It's a special kind of function. It is not a proper component, and I think in some places we used to use Storybook's own `withKnobs` (now deprecated) decorator, so I am using the same convention.

[the remark this answers is not in the document]
Comment (philip-cline) [reply packages/endpoints-overlay/src/EndpointsOverlay.story.tsx:40]: ok makes sense!

[the remark this answers is not in the document]
Comment (binh-dam-ibigroup) [reply packages/endpoints-overlay/src/EndpointsOverlay.story.tsx:40]: And I think that convention is used [here](https://storybook.js.org/tutorials/create-an-addon/react/en/decorators/), so I'll go ahead and rename to `withMap`.

File: packages/endpoints-overlay/src/EndpointsOverlay.story.tsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  </BaseMap>
+);
+
+export default {
+  component: EndpointsOverlay,
+  decorators: [mapDecorator],
+  title: "EndpointsOverlay"
+} as ComponentMeta<typeof EndpointsOverlay>;
+
+export const EndpointsOverlayWithoutUserSettings = (): ReactElement => (
```
[the remark this answers is not in the document]
Comment (binh-dam-ibigroup) [reply packages/endpoints-overlay/src/EndpointsOverlay.story.tsx]: Thanks for mentioning that, fixed in https://github.com/opentripplanner/otp-ui/pull/385/commits/e1819ff314d2b72cc0f515d35058a1debd140843.

File: packages/endpoints-overlay/src/EndpointsOverlay.story.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (binh-dam-ibigroup) [reply packages/endpoints-overlay/src/EndpointsOverlay.story.tsx]: Updated in https://github.com/opentripplanner/otp-ui/pull/385/commits/e1819ff314d2b72cc0f515d35058a1debd140843.

**Pull request merged**